### PR TITLE
refactor: refactor ddl manager

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

refactor ddl manager

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: src/common/meta/src/ddl_manager.rs [hunk 1]
```diff
@@ -14,18 +14,382 @@
...
+        self.submit_procedure(procedure_with_id).await
+    }
+
+    pub async fn submit_truncate_table_task(
+        &self,
+        cluster_id: u64,
+        truncate_table_task: TruncateTableTask,
+        region_routes: Vec<RegionRoute>,
+    ) -> Result<ProcedureId> {
+        error!("truncate table procedure is not supported, cluster_id = {}, truncate_table_task = {:?}, region_routes = {:?}",
```
Comment (fengjiachun) [inline src/common/meta/src/ddl_manager.rs:192]:
```suggestion
        error!("Truncate table procedure is not supported, cluster_id = {}, truncate_table_task = {:?}, region_routes = {:?}",
```

File: src/common/meta/src/ddl_manager.rs [hunk 2]
```diff
@@ -14,18 +14,382 @@
...
+                    let context = context.clone();
+                    AlterTableProcedure::from_json(json, context).map(|p| Box::new(p) as _)
+                }),
+            )
+            .context(RegisterProcedureLoaderSnafu {
+                type_name: AlterTableProcedure::TYPE_NAME,
+            })
+    }
+
+    pub async fn submit_alter_table_task(
```
Comment (fengjiachun) [inline src/common/meta/src/ddl_manager.rs:126]: I think we need to print some logs to record all DDL changes.
We can print info log at the beginning of handleXXX or submitXXX method.

File: src/meta-srv/src/metasrv.rs [hunk 3]
```diff
@@ -97,7 +97,7 @@ impl MetaSrvOptions {
     }
 }
 
-pub struct MetasrvInfo {
+pub struct MetaSrvInfo {
```
Comment (fengjiachun) [inline src/meta-srv/src/metasrv.rs:100]: It appears that we use `Metasrv` elsewhere.

**Review by fengjiachun — APPROVED** [review]

**Comment by WenyXu** [comment]
@fengjiachun PTAL

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**